### PR TITLE
Add new resource provider Microsoft.EngagementFabric

### PR DESCRIPTION
**Pull request by xzsheng**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.

The API spec was review and merged via [Private repo PR 569](https://github.com/Azure/azure-rest-api-specs-pr/pull/569)

### CODE REVIEW

File: specification/engagementfabric/resource-manager/Microsoft.EngagementFabric/preview/2018-09-01-preview/EngagementFabric.json [hunk 1]
```diff
@@ -0,0 +1,1280 @@
+{
+  "swagger": "2.0",
```
Comment (dsgouda) [inline specification/engagementfabric/resource-manager/Microsoft.EngagementFabric/preview/2018-09-01-preview/EngagementFabric.json]: Please rename directory `2018-09-01-preview` to `2018-09-01`

**Comment by dsgouda** [comment]
@xzsheng Please link the PR that was approved in the private repo
Were there any additional changes made to the specs since the PR was approved?

**Review by dsgouda — CHANGES_REQUESTED** [review]

File: specification/engagementfabric/resource-manager/readme.nodejs.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+``` yaml $(nodejs)
+nodejs:
+  azure-arm: true
+  package-name: azure-arm-engagementfabric
+  package-version: 1.0.0-preview
+  output-folder: $(node-sdks-folder)/lib/services/engagementFabricManagement
+  generate-license-txt: true
+  generate-package-json: true
+  generate-readme-md: true
```
Comment (daschult) [inline specification/engagementfabric/resource-manager/readme.nodejs.md]: Please also create a readme.typescript.md file. You can find several other packages in this repository that have readme.typescript.md files that you can copy and modify slightly to get the results you need.

**Comment by xzsheng** [comment]
> @xzsheng Please link the PR that was approved in the private repo
> Were there any additional changes made to the specs since the PR was approved?

@dsgouda The approved PR in private repo is https://github.com/Azure/azure-rest-api-specs-pr/pull/569
There is no change since the PR was approved.

File: specification/engagementfabric/resource-manager/Microsoft.EngagementFabric/preview/2018-09-01-preview/EngagementFabric.json (hunk 1 above)
Comment (xzsheng) [reply specification/engagementfabric/resource-manager/Microsoft.EngagementFabric/preview/2018-09-01-preview/EngagementFabric.json]: Fixed

File: specification/engagementfabric/resource-manager/readme.nodejs.md (hunk 2 above)
Comment (xzsheng) [reply specification/engagementfabric/resource-manager/readme.nodejs.md]: File added.

**Comment by dsgouda** [comment]
@jhendrixMSFT could you take a look at the Go generation failures, not sure why it's failing

**Comment by jhendrixMSFT** [comment]
From the log:
```
System.InvalidOperationException: codegen for preview swagger Microsoft.EngagementFabric/preview/2018-09-01/EngagementFabric.json must be under a preview subdirectory
```
Also I don't see the readme.go.md file.  Please be sure to put Go SDK config in its own config file not readme.md.

**Comment by dsgouda** [comment]
Thanks @jhendrixMSFT, @xzsheng please address the comments. You can refer to some existing readme.go.md files in the repo.

**Comment by xzsheng** [comment]
> Thanks @jhendrixMSFT, @xzsheng please address the comments. You can refer to some existing readme.go.md files in the repo.

@dsgouda Readme.go.md added. All checks are green now. Thanks.

**Review by dsgouda — APPROVED** [review]
LGTM

**Pull request merged**